spring-core/src/main/java/org/springframework/core/io/AbstractFileResolvingResource.java

```diff
@@ -298,6 +301,39 @@ protected void customizeConnection(HttpURLConnection con) throws IOException {
 		con.setRequestMethod("HEAD");
 	}
 
+	/**
+	 * This implementation returns the contents of a file as a string using the
+	 * system default Charset. Provided the resource exists and the context has
+	 * access to it, the contents will be returned as a single string with line
+	 * feed characters retained.
+	 * @return the contents of the requested file as a {@code String}.
+	 * @throws FileNotFoundException in the event the file path is invalid.
+	 * @throws IOException if the file can not be read or cannot be serialzied.
+	 */
+	@Override
+	public String getContentAsString() throws IOException {
+
+		if( !exists() ) {
+			throw new FileNotFoundException(getDescription() + " cannot be found.");
+		}
+		if ( !isReadable() ) {
+			throw new IOException(getDescription() + " cannot be opened for reading.");
+		}
+		return new String(Files.readAllBytes(Paths.get(getFile().getAbsolutePath())), Charset.defaultCharset());
```

Comment:
Relying on the default charset is a recipe for disaster, as it can vary between environments and make code fail in collocation that runs fine in another.

Because we cannot determine the resource's charset with a 100% certainty, `Resource::getContentAsString()` should be removed, or it should be defined to use UTF-8 as default. (see `Files::readString`)

Reply:
Thanks Poutsma! 

As this function is the core of this feature proposal I'll update the request to use utf-8 by default instead of throwing out the whole PR lol.